auto_nag/user_activity.py

```diff
@@ -93,16 +153,97 @@ def handler(user, data):
             + self.include_fields,
         ).wait()
 
-        return user_statuses
+        return users
+
+    def _get_status_from_phab_user(self, user: dict) -> UserStatus:
+        availability = user["attachments"]["availability"]
+        if availability["value"] != "available":
+            # We do not need to consider the user inactive they will be
+            # available again soon.
+            if (
+                not availability["until"]
+                or availability["until"] > self.availability_limit
+            ):
+                return UserStatus.UNAVAILABLE
+
+        return UserStatus.ACTIVE
+
+    def get_phab_users_with_status(
+        self, user_phids: List[str], keep_active: bool = False
+    ) -> dict:
+        """Get Phabricator users with their activity statuses.
+
+        Args:
+            user_phids: A list of user PHIDs.
+            keep_active: whether the returned results should include the active
+                users.
+
+        Returns:
+            A dictionary where the key is the user PHID and the value is
+            the user info with the status.
+        """
+
+        bzid_to_phid = {
+            int(user["id"]): user["phid"]
+            for _user_phids in Connection.chunks(user_phids, PHAB_CHUNK_SIZE)
+            for user in self._fetch_bz_user_ids(user_phids=_user_phids)
+        }
+        if not bzid_to_phid:
+            return {}
+
+        if "id" not in self.include_fields:
+            self.include_fields.append("id")
+
+        user_bz_ids = list(bzid_to_phid.keys())
+        users = self.get_bz_users_with_status(user_bz_ids, keep_active=True)
+        users = {bzid_to_phid[user["id"]]: user for user in users.values()}
+
+        # To cover cases where a person is temporary off (e.g., long PTO), we
+        # will rely on the calendar from phab.
+        for _user_phids in Connection.chunks(user_phids, PHAB_CHUNK_SIZE):
+            for phab_user in self._fetch_phab_users(_user_phids):
+                user = users[phab_user["phid"]]
+                if user["status"] == UserStatus.ACTIVE:
+                    user["status"] = self._get_status_from_phab_user(phab_user)
+
+                if not keep_active and user["status"] == UserStatus.ACTIVE:
+                    del users[phab_user["phid"]]
+                    continue
+
+                user["phab_username"] = phab_user["fields"]["username"]
+
+        return users
 
     def get_string_status(self, status: UserStatus):
+        """Get a string representation of the user status."""
+
         if status == UserStatus.UNDEFINED:
             return "Not specified"
-        elif status == UserStatus.DISABLED:
+        if status == UserStatus.DISABLED:
```

Comment:
Nit: why the change from elif to if?

Reply:
We have `return` for each statement, so the else is useless. I did it to satisfy Pylint.